prow/plugins/trigger/generic-comment_test.go

```diff
@@ -1481,3 +1482,41 @@ func TestRetestFilter(t *testing.T) {
 		})
 	}
 }
+
+func TestFailingWorkflow(t *testing.T) {
+	var testCases = []struct {
+		name                             string
+		triggerFailedGitHubWorkflowError error
+		runID                            int
+		expectedMessage                  string
+	}{
+		{
+			name:                             "Triggered workflow failed without err",
+			runID:                            1,
+			triggerFailedGitHubWorkflowError: errors.New(``),
+			expectedMessage:                  "",
+		},
+		{
+			name:                             "Triggered workflow failed due to expired workflow",
+			triggerFailedGitHubWorkflowError: errors.New(`{"message": "Unable to retry this workflow run because it was created over a month ago","documentation_url": "https://docs.github.com/rest/actions/workflow-runs#re-run-failed-jobs-from-a-workflow-run"}`),
+			runID:                            2,
+			expectedMessage:                  `{"message": "Unable to retry this workflow run because it was created over a month ago","documentation_url": "https://docs.github.com/rest/actions/workflow-runs#re-run-failed-jobs-from-a-workflow-run"}`,
+		},
+	}
+	c := fakegithub.NewFakeClient()
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			c.TriggerFailedGitHubWorkflowError = testCase.triggerFailedGitHubWorkflowError
```

Comment:
Hmm
I think let's create the client newly each time and assign the error value? We prevent races this way (right now we probably don't have any other test making use of `TriggerFailedGitHubWorkflow` but in the future there might be, and that might run in parallel, so let's avoid that).